runtime/lua/vim/diagnostic.lua

```diff
@@ -1535,8 +1535,12 @@ M.handlers.signs = {
       return
     end
 
+    if type(opts.signs) ~= 'table' then
+      return
+    end
```

Comment:
The old code was handling the case where `opts.signs` was not provided. If `signs=true` and the handler is called, then this handler should make default decisions, not skip itself entirely.

So maybe this change makes sense?:

```suggestion
    opts.signs = opts.signs or {}
```

Reply:
what code is calling this with `opts.signs=true`? Maybe it has a bug. Meanwhile, this is a 2-line fix:

    assert(opts.signs ~= false)
    opts.signs = type(opts.signs) == 'table' and opts.signs or {}

Reply:
No this is not needed. In `vim.diagnostic.show` we resolve the set of options so that when an option is set to `true` it is replaced by a default table.

The handlers should ALWAYS receive a table for `opts`. If they don't, it's a bug. Handlers do not need to check the type of `opts` or `opts.signs`. If they're getting something besides a table that's a bug, but not a bug in the handler.